graphql/resolvers.go

```diff
@@ -414,31 +430,36 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 		options.Expiration = spawnHostInput.Expiration
 	}
 
+	// passing an empty string taskId is okay as long as a
+	// taskId is not required by other spawnHostInput parameters
+	var t *task.Task
+	if spawnHostInput.TaskID != nil && *spawnHostInput.TaskID != "" {
```

Comment:
Is TaskID only passed in if one of those options is true? Cause then we can just check if t == nil once here

Reply:
I expect TaskId to be a string greater than length 0 if this condition is true. I compare TaskId against an empty string because I'd like to process empty strings and nil pointers the same way in the resolver. In our JS code, we often use an empty string to resemble empty or omitted form state so processing empty strings the same as nil pointers in the resolver allows us to propagate js form state directly into our gql queries without having to do any mutations to represent null or omitted values.

Reply:
Sorry what I meant was, can we check if t == nil at 440, instead of individually at 443, 448, and 458? I'm assuming if UseProjectSetupScript is set, then we _know_ taskId was provided, and same with TaskSync and UseProjectSetupScript.

Reply:
Ah right, it is possible to throw an error on line 440 when `t` is nil and we can reduce the amount of switching in this file. 
`t` could be a user inputted value. The benefit of checking whether `t` is nil after verifying that the dependent variable exists is so we can avoid an unnecessary error when `t` is nil and not utilized

Reply:
it is possible to execute this resolver code from any client that can make an http request to evergreen. The resolver parameters (like taskId which is used to intialize `t`) are usually provided from another script command which could be unknown to me. I could see it being helpful to our users if the api didn't completely error when an invalid and ununtilized parameter is supplied.